MODULE.bazel

```diff
@@ -1,2 +1,173 @@
-# TODO: migrate all dependencies from WORKSPACE to MODULE.bazel
-# https://github.com/bazelbuild/intellij/issues/5432
+module(
+    bazel_compatibility = [">=7.0.0"],
+    repo_name = "intellij_with_bazel",
+)
+
+bazel_dep(
+    name = "platforms",
+    version = "0.0.8",
+)
+bazel_dep(
+    name = "rules_go",
+    version = "0.46.0",
+    repo_name = "io_bazel_rules_go",
+)
+bazel_dep(
+    name = "rules_java",
+    version = "7.4.0",
+)
+bazel_dep(
+    name = "rules_python",
+    version = "0.31.0",
+)
+
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+python.toolchain(
+    configure_coverage_tool = True,
+    python_version = "3.12",
+)
+use_repo(python, "python_3_12", "python_versions")
+
+bazel_dep(
+    name = "rules_proto",
+    version = "5.3.0-21.7",
+)
+bazel_dep(
+    name = "protobuf",
+    version = "21.7",
+    repo_name = "com_google_protobuf",
+)
+bazel_dep(
+    name = "bazel_skylib",
+    version = "1.5.0",
+)
+bazel_dep(
+    name = "rules_pkg",
+    version = "0.10.1",
+)
+bazel_dep(
+    name = "rules_kotlin",
+    version = "1.9.0",
+)
+bazel_dep(
+    name = "rules_android",
+    version = "0.1.1",
+    repo_name = "build_bazel_rules_android",
+)
+
+remote_android_extensions = use_extension("@bazel_tools//tools/android:android_extensions.bzl", "remote_android_tools_extensions")
+use_repo(remote_android_extensions, "android_gmaven_r8", "android_tools")
+
+bazel_dep(
+    name = "rules_bazel_integration_test",
+    version = "0.23.0",
+    dev_dependency = True,
+)
+
+bazel_binaries = use_extension(
+    "@rules_bazel_integration_test//:extensions.bzl",
+    "bazel_binaries",
+    dev_dependency = True,
+)
+bazel_binaries.download(version = "4.2.4")
+bazel_binaries.download(version = "6.5.0")
+bazel_binaries.download(version = "7.0.2")
+use_repo(
+    bazel_binaries,
+    "bazel_binaries",
+    "bazel_binaries_bazelisk",
+    "build_bazel_bazel_4_2_4",
```

Comment:
I think the test for version 4.2.4 is missing: https://buildkite.com/bazel/intellij-plugin-aspect/builds/21280#018dd20e-4bca-481d-8d9b-060f884523f5. 

We actually need to test with a version before 6 and a version after it because based on that we change the label of the aspect repo in the sync invocation.